apps/src/applab/AppLabView.jsx

```diff
@@ -173,8 +164,16 @@ var AppLabView = React.createClass({
     return (
       <ConnectedStudioAppWrapper>
         <div id="visualizationColumn">
-          {playSpaceHeader}
-          <ProtectedStatefulDiv contentFunction={this.props.generateVisualizationColumnHtml} />
+          {!this.props.isReadOnlyWorkspace && <PlaySpaceHeader
+              isEditingProject={this.props.isEditingProject}
+              screenIds={this.props.screenIds}
+              onViewDataButton={this.props.onViewDataButton}
+              onScreenCreate={this.props.onScreenCreate} />
+          }
+          <ApplabVisualizationColumn
+              isProjectLevel={this.props.isProjectLevel}
+              isSubmittable={this.props.isSubmittable}
+              isSubmitted={this.props.isSubmitted}/>
```

Comment:
To be specific to this case, I would totally connect `AppLabVisualizationColumn` so that its three props are no longer listed anywhere in `AppLabView`, because that makes it clear that this layer is not concerned with those properties.